src/ensembl/src/shared/helpers/utils.test.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import faker from 'faker';
+
+import { omit } from './utils';
+
+describe('omit', () => {
+  it('removes the passed in key from the source object', () => {
+    const key = faker.lorem.word();
+    const key1 = faker.lorem.word();
+    const key2 = faker.lorem.word();
+
+    const obj = {
+      [key]: {
+        [key1]: {
+          [key2]: {
+            foo: 1,
+            bar: 1
+          }
+        }
+      }
+    };
+
+    const updateObj = {
+      [key]: {
+        [key1]: {
+          [key2]: {
+            bar: 1
+          }
+        }
+      }
+    };
+
+    expect(omit(obj, [key, key1, key2, 'foo'])).toEqual(updateObj);
```

Comment:
Mmm, the signature of your `omit` is similar to lodash's, but the behaviour is different.

Normally, `omit` (such as in lodash or in ramda, or in the typescript `Omit` utility type) will take an object and an array of keys, and return an object without the passed keys:

```js
// from lodash docs:
var object = { 'a': 1, 'b': '2', 'c': 3 };
 
_.omit(object, ['a', 'c']);
// => { 'b': '2' }
``` 

What you wrote is an immutable unset rather than an omit 😄

Reply:
Oh yea! my bad 😄  I've updated it now.